fearless/Modules/Staking/ControllerAccount/ControllerAccountPresenter.swift

```diff
@@ -0,0 +1,242 @@
+            return
+        }
+
+        guard let accounts = accounts, let chosenAccountItem = chosenAccountItem else {
+            return
+        }
+        let context = PrimitiveContextWrapper(value: accounts)
+        let title = LocalizableResource<String> { locale in
+            R.string.localizable
+                .stakingControllerAccountTitle(preferredLanguages: locale.rLanguages)
+        }
+        wireframe.presentAccountSelection(
+            accounts,
+            selectedAccountItem: chosenAccountItem,
+            title: title,
+            delegate: self,
+            from: view,
+            context: context
+        )
+    }
+
+    func handleStashAction() {
+        presentAccountOptions(for: stashAccount?.address)
+    }
+
+    private func presentAccountOptions(for address: AccountAddress?) {
+        guard
+            let view = view,
+            let address = address
+        else { return }
+        wireframe.presentAccountOptions(
+            from: view,
+            address: address,
+            chain: chain,
+            locale: view.localizationManager?.selectedLocale ?? .current
+        )
+    }
+
+    func selectLearnMore() {
+        guard let view = view else { return }
+        wireframe.showWeb(
+            url: applicationConfig.learnControllerAccountURL,
+            from: view,
+            style: .automatic
+        )
+    }
+
+    func proceed() {
+        let locale = view?.localizationManager?.selectedLocale ?? Locale.current
+        DataValidationRunner(validators: [
+            dataValidatingFactory.has(fee: fee, locale: locale, onError: { [weak self] in
+                self?.refreshFeeIfNeeded()
+            }),
+            dataValidatingFactory.canPayFee(
+                balance: balance,
+                fee: fee,
+                locale: locale
+            ),
+            dataValidatingFactory.ledgerNotExist(
+                stakingLedger: stakingLedger,
+                for: chosenAccountItem?.address,
+                addressType: chain.addressType,
+                locale: locale
+            )
+        ]).runValidation { [weak self] in
+            self?.wireframe.showConfirmation(from: self?.view)
+        }
+    }
+}
+
+extension ControllerAccountPresenter: ControllerAccountInteractorOutputProtocol {
+    func didReceiveStashItem(result: Result<StashItem?, Error>) {
+        switch result {
+        case let .success(stashItem):
+            self.stashItem = stashItem
+            updateView()
+            if stashItem == nil {
+                wireframe.close(view: view)
+            }
+        case let .failure(error):
+            logger?.error("Did receive stash item error: \(error)")
+        }
+    }
+
+    func didReceiveStashAccount(result: Result<AccountItem?, Error>) {
+        switch result {
+        case let .success(accountItem):
+            stashAccount = accountItem
+            chosenAccountItem = accountItem
```

Comment:
Why stash? We are choosing a controller

Reply:
We set initial value of chosenAccountItem to stashAccountItem